### DISCUSSION TITLE
delete selected region

### DISCUSSION THREAD

**Discussion by prtj** [katspaugh/wavesurfer.js#3113] [Q&A] [answered]
<img width="875" alt="截屏2023-08-14 15 41 08" src="https://github.com/katspaugh/wavesurfer.js/assets/122947574/1ada3271-1f13-4e57-b183-517ca5e81b7f">
I am using WaveSurfer 7.1.2 ，and I want to delete this highlighted area with the right mouse button, but I don't know how to implement it

**Comment by katspaugh** [comment]
You can pass a custom element in the `content` property of a region, and assign a right click handler on that element. In the handler, call `region.remove()`.

**Reply by prtj** [reply]
But I can't write content on my region,could I can use the following code？
this.wavesurfer = WaveSurfer.create({
        container: this.$refs.waveform,
        waveColor: 'rgb(200, 0, 200)',
        url: require('./audio_1.mp3'),
        plugins:[
        ]
      });
      
this.regions = this.wavesurfer.registerPlugin(Regions.create())
this.wavesurfer.setVolume(0.0);
this.regions.enableDragSelection({
  color: 'rgb(173,216,230,0.5)',
});
this.wavesurfer.on('region-contextmenu', function (region, event) {
  if (confirm('Delete？')) {
    region.remove();
  }
  event.preventDefault();
});

**Reply by katspaugh** [reply, marked as answer]
Actually you don't need a custom element, you can listen to clicks on the region itself.

```js
wsRegions.enableDragSelection({
  color: 'rgb(173,216,230,0.5)',
})

wsRegions.on('region-clicked', (region, e) => {
  if (e.button === 2) {
    region.remove()
  }
})
```

Please refer to the docs for any further details: https://wavesurfer-js.org/docs/modules/plugins_regions

**Comment by prtj** [comment]
this.regions.on('region-clicked',(region,e)=>{  
        console.log("region_clicked")
        console.log(e)
        if (this.activeRegion !== null &&  this.activeRegion.id != region.id)  {
          this.activeRegion.setOptions({
          color: 'rgb(173,216,230,0.5)',
        })
        }
        this.activeRegion = region;
        if (e.button === 2) {
          region.remove()
        }
        e.stopPropagation();
        this.currentTime = region.start;
        audio.currentTime = this.currentTime;
        region.setOptions({
          color: 'rgb(255,228,225,0.5)'
        });
        if (this.audio_is_play){
          region.play(); 

        }  
    });
<img width="425" alt="截屏2023-08-14 17 21 00" src="https://github.com/katspaugh/wavesurfer.js/assets/122947574/7cbc76a9-83b0-4390-9140-d696462ae12a">
I have tried this before, but after I clicked on region, right-click again and it will appear this

**Reply by katspaugh** [reply]
Add `e.preventDefault()`.
